### ISSUE TITLE
Building on mac gives errors on certain symbols 

### ISSUE THREAD

**Issue by bolkedebruin** [msteinert/pam#24]
```
../../go/pkg/mod/github.com/msteinert/pam/v2@v2.0.0/errors.go:81:21: could not determine kind of name for C.PAM_BAD_ITEM
../../go/pkg/mod/github.com/msteinert/pam/v2@v2.0.0/errors.go:84:23: could not determine kind of name for C.PAM_CONV_AGAIN
../../go/pkg/mod/github.com/msteinert/pam/v2@v2.0.0/errors.go:88:24: could not determine kind of name for C.PAM_INCOMPLETE
```

They are probably Linux specific. Can these be OS architecture dependent?

**Comment by 3v1n0** [comment]
Yeah, it could be definitely be the case... We need check if those are defined and provide fallbacks otherwise.

**Comment by 3v1n0** [comment]
Maybe we should move these errors to linux-specific?

so a `errors_linux.go` with:

```
//go:build linux
// These errors are only linux specific.
package pam

const (
	// ErrBadItem indicates a bad item passed to pam_*_item().
	ErrBadItem Error = C.PAM_BAD_ITEM
	// ErrConvAgain indicates a conversation function is event driven and data
	// is not available yet.
	ErrConvAgain Error = C.PAM_CONV_AGAIN
	// ErrIncomplete indicates to please call this function again to complete
	// authentication stack. Before calling again, verify that conversation
	// is completed.
	ErrIncomplete Error = C.PAM_INCOMPLETE
)
```

Although `BAD_ITEM` seems to be there in freebsd as per #30, so maybe it's just better to redefine them in case they're not found at C-level, but that would make the symbols being available in platforms where they should not (not a big deal I guess).

In any case I'd have to figure out how to adapt tests in this case, or just make them linux-specific too.

**Comment by msteinert** [comment]
I think the solution for this and #30 is to create platform-specific definitions using the `//go:build` tag (same for any platform-specific tests). Ideally we would then have CI jobs that run on the supported platforms to prevent any future problems...

**Comment by VVD** [comment]
Forgejo update in FreeBSD ports is delayed due to build error in pam v2: https://bugs.freebsd.org/286296

**Comment by msteinert** [comment]
@3v1n0 It looks like the error definitions diverge between implementations:

LInux-PAM:
https://github.com/linux-pam/linux-pam/blob/master/libpam/include/security/_pam_types.h#L76-L91

OpenPAM:
https://git.des.dev/OpenPAM/OpenPAM/src/branch/main/include/security/pam_constants.h#L78-L83

**Comment by msteinert** [comment]
It also appears that there is no support for `pam_start_confdir` in OpenPAM.

**Comment by msteinert** [comment]
I threw together a quick PR to get this build and run the test suite on my MacBook: https://github.com/msteinert/pam/pull/31

I don't have time to spin up a FreeBSD VM right now, if somebody wants to test it out there and comment on the PR I would appreciate it.

**Comment by VVD** [comment]
I applied this patch and Forgejo built without errors on FreeBSD 14.2 amd64.
Runtime not tested.

**Comment by 3v1n0** [comment]
> It also appears that there is no support for `pam_start_confdir` in OpenPAM.

Indeed but that was already checked.
